src/api/app/assets/javascripts/webui2/buildresult.js

```diff
@@ -35,6 +36,39 @@ function updateBuildResult(index) { // jshint ignore:line
   });
 }
 
+// TODO: Handle 'ajax:error', 'ajax:complete' and DRY binding code
+function subscribeAjaxEvents(index) {
+  $('#show_all_' + index).on('ajax:success', function(_event, data, _status, _xhr) {
+    $('#build' + index + ' .result').html(data);
+    // Since the element on which we bind this event is rendered again on the line just above, it needs to be bound again...
```

Comment:
You can avoid this by attaching a generic event handler to the element that contains all the build results, e.g.
```js
$('.build-result-table').on('ajax:success', '.show-all', function() {
...
```

The index would get moved away to the data attribute of the `.show-all` element, if it's even needed. You probably could replace the `$('#build' + index + ' .result').html(data);` with something like `$(this).find('.build-result')` or `$(this).children`, sibling...

Reply:
In the end, I won't be using this function at all.